pkg/binary/envoy/runtime.go

```diff
@@ -130,49 +141,33 @@ func (r *Runtime) GetPid() (int, error) {
 	return r.cmd.Process.Pid, nil
 }
 
-func (r *Runtime) envoyReady() bool {
+// envoyReady reads the HTTP status from the /ready endpoint and returns binary.StatusReady on 200 or
+// binary.StatusInitializing on 503.
+func (r *Runtime) envoyReady() (int, error) {
+	adminAddress := r.GetAdminAddress()
+	if adminAddress == "" { // don't yet have an admin address
+		return binary.StatusInitializing, nil
+	}
+
 	// Once we have seen its ready once stop spamming the ready endpoint.
 	// If we expand the interface to support ready <-> not ready then
 	// this approach will be wrong but as the states are monotonic this is good enough for now
 	if r.isReady {
-		return true
+		return binary.StatusReady, nil
 	}
-	resp, err := http.Get(fmt.Sprintf("http://%s/ready", r.Config.GetAdminAddress()))
+	resp, err := http.Get(fmt.Sprintf("http://%s/ready", adminAddress))
 	if err != nil {
-		return false
+		return 0, err
 	}
 	defer resp.Body.Close() //nolint
-	if resp.StatusCode == http.StatusOK {
+	switch resp.StatusCode {
+	case http.StatusOK:
 		r.isReady = true
-		return r.isReady
-	}
-	return false
-}
-
-// Wait blocks until the child process reaches the state passed
-// Note: It does not guarantee that it is in the specified state just that it has reached it
-func (r *Runtime) Wait(state int) {
-	for r.Status() < state {
-		// This is a call to a function to allow the goroutine to be preempted for garbage collection
-		// The sleep duration is somewhat arbitrary
-		func() { time.Sleep(time.Millisecond * 100) }()
-	}
-}
-
-// WaitWithContext blocks until the child process reaches the state passed or the context is canceled
-// Note: It does not guarantee that it is in the specified state just that it has reached it
-func (r *Runtime) WaitWithContext(ctx context.Context, state int) {
```

Comment:
the wait function was only used once EnableEnvoyLogCollection, so moved it there